src/main/java/org/numenta/nupic/network/Layer.java

```diff
@@ -1909,12 +1913,35 @@ private void clearSubscriberObserverLists() {
      * @return
      */
     NamedTuple makeClassifiers(MultiEncoder encoder) {
+        Map inferredFields = (Map<String, Class<? extends Classifier>>) params.get(KEY.INFERRED_FIELDS);
+        if(inferredFields == null || inferredFields.entrySet().size() == 0) {
+            throw new IllegalStateException(
+                    "KEY.AUTO_CLASSIFY has been set to \"true\", but KEY.INFERRED_FIELDS is null or\n\t" +
+                    "empty. Must specify desired Classifier for at least one input field in\n\t" +
+                    "KEY.INFERRED_FIELDS or set KEY.AUTO_CLASSIFY to \"false\"."
```

Comment:
@Hopding I would rather not specify KEY.AUTO_CLASSIFY at all, rather than force the user to set it to false. It adds verboseness where it isn't needed? You should probably change this to, "Must specify desired Classifier for at least one input field in "KEY.INFERRED_FIELDS" or not specify "KEY.AUTO_CLASSIFY" at all."